### ISSUE TITLE
[rarbg] (testing) Exception (rarbg): The response was not JSON: The response was not JSON

### ISSUE THREAD

**Issue by zerosquared** [Jackett/Jackett#13358]
### Have you checked our Troubleshooting page for your issue?

- [X] I have checked the Troubleshooting page

### Is there already an issue for your problem?

- [X] I have checked older issues, open and closed

### Have you read our Contributing Guidelines?

- [X] I have read the Contributing Guidelines

### Environment

```markdown
2022-07-13 16:34:25.2602 Info Starting Jackett v0.20.1304 
2022-07-13 16:34:25.9720 Info Environment version: 6.0.7 (C:\ProgramData\Jackett\) 
2022-07-13 16:34:25.9720 Info OS version: Microsoft Windows NT 10.0.19043.0 (64bit OS) 
2022-07-13 16:34:25.9720 Info Jackett variant: CoreWindows 
2022-07-13 16:34:25.9720 Info Running in Docker: No 
2022-07-13 16:34:25.9720 Info ThreadPool MaxThreads: 1023 workerThreads, 1000 completionPortThreads 
2022-07-13 16:34:25.9720 Info App config/log directory: C:\ProgramData\Jackett 
2022-07-13 16:34:25.9720 Info Using proxy: Disabled 
2022-07-13 16:34:25.9720 Info Using FlareSolverr: No 
2022-07-13 16:34:25.9812 Info Using HTTP Client: HttpWebClient2
```

### Description

This just started a couple of days ago where RARBG is throwing errors in Sonarr.  I removed it from Jackett and re-added it and Sonarr was fine for about 10 seconds before showing failure again.  Not sure what else to try as I'm not a programmer, but this program's been a life saver.

Test Results from Jackett:
An error occurred while testing this indexer
Exception (rarbg): The response was not JSON: The response was not JSON

[log.txt](https://github.com/Jackett/Jackett/files/9105983/log.txt)

### Logged Error Messages

```text
2022-07-13 16:34:52.5429 Error Jackett.Common.IndexerException: Exception (rarbg): The response was not JSON
 ---> System.Exception: The response was not JSON
   at Jackett.Common.Indexers.RarBG.PerformQueryWithRetry(TorznabQuery query, Boolean retry) in D:\a\1\s\src\Jackett.Common\Indexers\RarBG.cs:line 148
   at Jackett.Common.Indexers.RarBG.PerformQuery(TorznabQuery query) in D:\a\1\s\src\Jackett.Common\Indexers\RarBG.cs:line 134
   at Jackett.Common.Indexers.BaseIndexer.ResultsForQuery(TorznabQuery query, Boolean isMetaIndexer) in D:\a\1\s\src\Jackett.Common\Indexers\BaseIndexer.cs:line 310
   --- End of inner exception stack trace ---
   at Jackett.Common.Indexers.BaseIndexer.ResultsForQuery(TorznabQuery query, Boolean isMetaIndexer) in D:\a\1\s\src\Jackett.Common\Indexers\BaseIndexer.cs:line 322
   at Jackett.Common.Indexers.BaseWebIndexer.ResultsForQuery(TorznabQuery query, Boolean isMetaIndexer) in D:\a\1\s\src\Jackett.Common\Indexers\BaseIndexer.cs:line 677
   at Jackett.Common.Services.IndexerManagerService.TestIndexer(String name) in D:\a\1\s\src\Jackett.Common\Services\IndexerManagerService.cs:line 309
   at Jackett.Server.Controllers.IndexerApiController.Test() in D:\a\1\s\src\Jackett.Server\Controllers\IndexerApiController.cs:line 132
   at Microsoft.AspNetCore.Mvc.Infrastructure.ActionMethodExecutor.TaskOfIActionResultExecutor.Execute(IActionResultTypeMapper mapper, ObjectMethodExecutor executor, Object controller, Object[] arguments)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeActionMethodAsync>g__Awaited|12_0(ControllerActionInvoker invoker, ValueTask`1 actionResultValueTask)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeNextActionFilterAsync>g__Awaited|10_0(ControllerActionInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Rethrow(ActionExecutedContextSealed context)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.Next(State& next, Scope& scope, Object& state, Boolean& isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ControllerActionInvoker.<InvokeInnerFilterAsync>g__Awaited|13_0(ControllerActionInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeFilterPipelineAsync>g__Awaited|20_0(ResourceInvoker invoker, Task lastTask, State next, Scope scope, Object state, Boolean isCompleted)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeAsync>g__Awaited|17_0(ResourceInvoker invoker, Task task, IDisposable scope)
   at Microsoft.AspNetCore.Mvc.Infrastructure.ResourceInvoker.<InvokeAsync>g__Awaited|17_0(ResourceInvoker invoker, Task task, IDisposable scope)
   at Microsoft.AspNetCore.Routing.EndpointMiddleware.<Invoke>g__AwaitRequestTask|6_0(Endpoint endpoint, Task requestTask, ILogger logger)
   at Microsoft.AspNetCore.Authentication.AuthenticationMiddleware.Invoke(HttpContext context)
   at Jackett.Server.Middleware.CustomExceptionHandler.Invoke(HttpContext httpContext) in D:\a\1\s\src\Jackett.Server\Middleware\CustomExceptionHandler.cs:line 26
```

**Comment by ilike2burnthing** [comment]
https://github.com/Jackett/Jackett/issues/13350#issuecomment-1179783543
https://github.com/Jackett/Jackett/commit/86d7a55329028977431830d6628ab2b4b8da2eb7
https://github.com/Jackett/Jackett/issues/13356#issuecomment-1182238808

TL;DR - reconfigure the RARBG indexer to increase the retry count

**Comment by Airilk** [comment]
I have the same issue with the OP. I get the same "Exception (rarbg): The response was not JSON: The response was not JSON" error. The increase of the retry count does not seem to be the solution. Is there another way to fix this? i am not a programmer either.

**Comment by ilike2burnthing** [comment]
Increasing the number of retries doesn't fix the issue, as the problem lies with RARBG, it only reduces the chance of a failure for a single search (e.g. failing 5 times in a row is much less likely, but still possible).

**Comment by Locutus64** [comment]
I'm experiencing the exact same issue with RARBG.

**Comment by ilike2burnthing** [comment]
See instructions above.
